pkg/endpoint/policy.go

```
@@ -913,6 +990,14 @@ func (e *Endpoint) SetIdentity(owner Owner, identity *identityPkg.Identity) {
 
 	e.runIdentityToK8sPodSync()
 
+	// Whenever the identity is updated, propagate change to key-value store
+	// of IP to identity mapping.
+
+	if e.IPv4 != nil {
```

This check could be done in `runIPv4IdentitySync` itself, but I'm not sure if it's better for it to exist here. If you did move it to `runIPv4IdentitySync` you could return an error to make it more obvious.

---

Similarly, perhaps we should also check IPv6? I'm not sure if it can ever be nil.

---

I'll add a nil check at the top of the function (which I have made generic for IPv4 and IPv6 per your recommendation); I don't think an error is necessary, as it is expected at times that there is no IPv4 address for an endpoint based on daemon configuration.

---

Done.